firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -821,13 +829,15 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
         }
     }
 
-    private func reAddTabs(tabsToAdd: [Tab], previousTabUUID: String) {
+    private func reAddTabs(tabsToAdd: [Tab], previousTabUUID: String, isPrivate: Bool = false) {
         tabs.append(contentsOf: tabsToAdd)
         let tabToSelect = tabs.first(where: { $0.tabUUID == previousTabUUID })
         let currentlySelectedTab = selectedTab
         if let tabToSelect = tabToSelect, let currentlySelectedTab = currentlySelectedTab {
-            // remove currently selected tab
-            removeTabs([currentlySelectedTab])
+            // remove currently selected tab only for normal mode
```

Comment:
Is this comment correct or am I just confused? It seems like currently selected tab is removed only if it's Private?

Reply:
Some of the comments could be parred down here if you wanted, the code is pretty readable via function names and it's less likely to get out-dated without comments, but I know it can be a case of personal preference too 🤷‍♀️

Reply:
thanks for the catch! the comment is correct, the code was incorrect. I added more details for the comment to explain why. Do you think its useful now? If not, I can delete it!

Reply:
It looks good now!